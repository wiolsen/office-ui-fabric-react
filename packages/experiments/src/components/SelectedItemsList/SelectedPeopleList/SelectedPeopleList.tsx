/* tslint:disable */
import * as React from 'react';
/* tslint:enable */
import { SelectedItemsList } from '../SelectedItemsList';
import { SelectedPersona } from './Items/SelectedPersona';
import { ISelectedItemsListProps, ISelectedItemsList, BaseSelectedItem } from '../SelectedItemsList.types';
import { IPersonaProps } from 'office-ui-fabric-react/lib/Persona';

type PartiallyOptional<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>> & Pick<Partial<T>, keyof T>;
export type ISelectedPeopleListProps<TPersona> = PartiallyOptional<ISelectedItemsListProps<TPersona>, 'onRenderItem'>;

export type ISelectedPeopleList<TPersona extends IPersonaProps & BaseSelectedItem = IPersonaProps> = ISelectedItemsList<TPersona>;
export type SelectedPeopleList<TPersona extends IPersonaProps & BaseSelectedItem = IPersonaProps> = SelectedItemsList<TPersona>;
export const SelectedPeopleList = React.forwardRef(
  <TPersona extends IPersonaProps & BaseSelectedItem = IPersonaProps>(
    props: ISelectedPeopleListProps<TPersona>,
    ref: React.Ref<ISelectedPeopleList<TPersona>>
  ) => <SelectedItemsList<TPersona> ref={ref} onRenderItem={SelectedPersona} {...props} />
);
