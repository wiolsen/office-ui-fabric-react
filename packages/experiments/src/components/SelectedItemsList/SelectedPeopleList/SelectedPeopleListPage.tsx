import * as React from 'react';
import { ExampleCard, IComponentDemoPageProps, ComponentPage, PropertiesTableSet } from '@uifabric/example-app-base';

import { SelectedPeopleListBasicExample } from './examples/SelectedPeopleList.Basic.Example';
const SelectedPeopleListBasicExampleCode = require('!raw-loader!./examples/SelectedPeopleList.Basic.Example') as string;

import { SelectedPeopleListWithEditInContextMenuExample } from './examples/SelectedPeopleList.WithEditInContextMenu.Example';
const SelectedPeopleListWithEditInContextMenuExampleCode = require('!raw-loader!./examples/SelectedPeopleList.WithEditInContextMenu.Example') as string;

import { SelectedPeopleListWithGroupExpandExample } from './examples/SelectedPeopleList.WithGroupExpand.Example';
const SelectedPeopleListWithGroupExpandExampleCode = require('!raw-loader!./examples/SelectedPeopleList.WithGroupExpand.Example') as string;

import { SelectedPeopleListWithEditExample } from './examples/SelectedPeopleList.WithEdit.Example';
const SelectedPeopleListWithEditExampleCode = require('!raw-loader!./examples/SelectedPeopleList.WithEdit.Example') as string;

import { SelectedPeopleListWithContextMenuExample } from './examples/SelectedPeopleList.WithContextMenu.Example';
const SelectedPeopleListWithContextMenuExampleCode = require('!raw-loader!./examples/SelectedPeopleList.WithContextMenu.Example') as string;

export class SelectedPeopleListPage extends React.Component<IComponentDemoPageProps, {}> {
  public render(): JSX.Element {
    return (
      <ComponentPage
        title="SelectedPeopleList"
        componentName="SelectedPeopleList"
        exampleCards={
          <div>
            <ExampleCard title="Basic" isOptIn={true} code={SelectedPeopleListBasicExampleCode}>
              <SelectedPeopleListBasicExample />
            </ExampleCard>
            <ExampleCard
              title="With Copying, Editing, and Expansion"
              isOptIn={true}
              code={SelectedPeopleListWithEditInContextMenuExampleCode}
            >
              <SelectedPeopleListWithEditInContextMenuExample />
            </ExampleCard>
            <ExampleCard title="With A Context Menu" isOptIn={true} code={SelectedPeopleListWithContextMenuExampleCode}>
              <SelectedPeopleListWithContextMenuExample />
            </ExampleCard>
            <ExampleCard title="With Expandable Groups" isOptIn={true} code={SelectedPeopleListWithGroupExpandExampleCode}>
              <SelectedPeopleListWithGroupExpandExample />
            </ExampleCard>
            <ExampleCard title="With Editing" isOptIn={true} code={SelectedPeopleListWithEditExampleCode}>
              <SelectedPeopleListWithEditExample />
            </ExampleCard>
            <ExampleCard title="With Edit as an option in the context menu" isOptIn={true} code={SelectedPeopleListWithEditExampleCode}>
              <SelectedPeopleListWithEditInContextMenuExample />
            </ExampleCard>
          </div>
        }
        propertiesTables={
          <PropertiesTableSet sources={[require<string>('!raw-loader!@uifabric/experiments/src/components/Tile/Tile.types.ts')]} />
        }
        overview={<div />}
        bestPractices={<div />}
        dos={
          <div>
            <ul>
              <li>Use them to display a list of picked people</li>
            </ul>
          </div>
        }
        donts={
          <div>
            <ul>
              <li>Use them to display things that aren't people</li>
            </ul>
          </div>
        }
        isHeaderVisible={this.props.isHeaderVisible}
      />
    );
  }
}
