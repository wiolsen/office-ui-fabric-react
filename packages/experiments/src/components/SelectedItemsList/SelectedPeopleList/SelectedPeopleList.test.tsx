import * as React from 'react';
import { act, create } from 'react-test-renderer';

import { SelectedPeopleList, ISelectedPeopleList } from './SelectedPeopleList';

describe('SelectedPeopleList', () => {
  it('renders personas that are passed in', () => {
    const pickerRef: React.RefObject<ISelectedPeopleList> = React.createRef();
    const rendered = create(<SelectedPeopleList ref={pickerRef} />);
    act(() => {
      pickerRef.current &&
        pickerRef.current.addItems([
          {
            text: 'Person A'
          },
          {
            text: 'Person B'
          }
        ]);
    });

    const personANodes = rendered.root.findAll(x => !!x.children.length && x.children.indexOf('Person A') !== -1);
    const personBNodes = rendered.root.findAll(x => !!x.children.length && x.children.indexOf('Person B') !== -1);
    expect(personANodes.length).toBeGreaterThan(0);
    expect(personBNodes.length).toBeGreaterThan(0);

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
