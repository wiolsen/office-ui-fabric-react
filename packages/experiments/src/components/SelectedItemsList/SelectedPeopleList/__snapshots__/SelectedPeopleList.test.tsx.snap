// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedPeopleList renders personas that are passed in 1`] = `
Array [
  <div
    aria-labelledby="selectedItemPersona-id__0"
    className=
        ms-PickerPersona-container
        {
          align-items: center;
          background: #eff6fc;
          border-radius: 15px;
          cursor: default;
          display: inline-flex;
          margin-bottom: 4px;
          margin-left: 4px;
          margin-right: 4px;
          margin-top: 4px;
          position: relative;
          user-select: none;
          vertical-align: middle;
        }
        &:hover {
          background: #deecf9;
        }
        @media screen and (-ms-high-contrast: active){&:hover {
          color: HighlightText;
        }
        @media screen and (-ms-high-contrast: active){& {
          -ms-high-contrast-adjust: none;
          background: Highlight;
          border-color: Highlight;
        }
    data-is-focusable={true}
    data-is-sub-focuszone={true}
    data-selection-index={0}
    role="listitem"
  >
    <div
      hidden={true}
    >
      <button
        className=
            ms-Button
            ms-Button--icon
            ms-PickerItem-removeButton
            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              background-color: transparent;
              border-radius: 15px 0px 0px 15px;
              border: none;
              box-sizing: border-box;
              color: #ffffff;
              cursor: pointer;
              display: flex;
              font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
              font-size: 14px;
              font-weight: 400;
              height: 33px;
              margin-right: -17px;
              outline: transparent;
              padding-bottom: 0;
              padding-left: 4px;
              padding-right: 16px;
              padding-top: 0;
              position: inherit;
              text-align: center;
              text-decoration: none;
              user-select: none;
              vertical-align: top;
              width: 44px;
            }
            &::-moz-focus-inner {
              border: 0;
            }
            .ms-Fabric--isFocusVisible &:focus:after {
              border: 1px solid transparent;
              bottom: 2px;
              content: "";
              left: 2px;
              outline: 1px solid #605e5c;
              position: absolute;
              right: 2px;
              top: 2px;
              z-index: 1;
            }
            @media screen and (-ms-high-contrast: active){.ms-Fabric--isFocusVisible &:focus:after {
              border: none;
              bottom: -2px;
              left: -2px;
              outline-color: ButtonText;
              right: -2px;
              top: -2px;
            }
            &:active > * {
              left: 0px;
              position: relative;
              top: 0px;
            }
            &:hover {
              background-color: #c7e0f4;
              color: #106ebe;
            }
            @media screen and (-ms-high-contrast: active){&:hover {
              border-color: Highlight;
              color: Highlight;
            }
            &:active {
              background-color: #edebe9;
              color: #005a9e;
            }
        data-is-focusable={true}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <div
          className=
              ms-Button-flexContainer
              {
                align-items: center;
                display: flex;
                flex-wrap: nowrap;
                height: 100%;
                justify-content: center;
              }
        >
          <i
            aria-hidden={true}
            className=
                ms-Button-icon
                {
                  -moz-osx-font-smoothing: grayscale;
                  -webkit-font-smoothing: antialiased;
                  color: #005a9e;
                  display: inline-block;
                  flex-shrink: 0;
                  font-family: "FabricMDL2Icons";
                  font-size: 16px;
                  font-style: normal;
                  font-weight: normal;
                  height: 16px;
                  line-height: 16px;
                  margin-bottom: 0;
                  margin-left: 4px;
                  margin-right: 4px;
                  margin-top: 0;
                  speak: none;
                  text-align: center;
                  vertical-align: middle;
                }
                @media screen and (-ms-high-contrast: active){& {
                  color: HighlightText;
                }
            data-icon-name="Add"
            role="presentation"
            style={
              Object {
                "fontSize": "14px",
              }
            }
          >
            Óúê
          </i>
        </div>
      </button>
    </div>
    <div
      className=

          {
            display: inherit;
            position: relative;
          }
    >
      <div
        className=
            ms-PickerItem-content
            {
              flex: 0 1 auto;
              max-width: 100%;
              min-width: 0px;
            }
        id="selectedItemPersona-id__0"
      >
        <div
          className=
              ms-Persona
              ms-Persona--size32
              {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                align-items: center;
                box-shadow: none;
                box-sizing: border-box;
                color: #323130;
                display: flex;
                font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                font-size: 14px;
                font-weight: 400;
                height: 32px;
                margin-bottom: 0px;
                margin-left: 0px;
                margin-right: 0px;
                margin-top: 0px;
                min-width: 32px;
                padding-bottom: 0px;
                padding-left: 0px;
                padding-right: 0px;
                padding-top: 0px;
                position: relative;
              }
              & .contextualHost {
                display: none;
              }
              &:hover $primaryText {
                color: #201f1e;
              }
        >
          <div
            className=
                ms-Persona-coin
                ms-Persona--size32
                {
                  -moz-osx-font-smoothing: grayscale;
                  -webkit-font-smoothing: antialiased;
                  font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                  font-size: 14px;
                  font-weight: 400;
                }
          >
            <div
              className=
                  ms-Persona-imageArea
                  {
                    flex: 0 0 auto;
                    height: 32px;
                    position: relative;
                    text-align: center;
                    width: 32px;
                  }
            >
              <div
                aria-hidden="true"
                className=
                    ms-Persona-initials
                    {
                      background-color: #69797E;
                      border-radius: 50%;
                      color: #ffffff;
                      font-size: 14px;
                      font-weight: 600;
                      height: 32px;
                      line-height: 32px;
                    }
                    @media screen and (-ms-high-contrast: active){& {
                      -ms-high-contrast-adjust: none;
                      background-color: Window !important;
                      border: 1px solid WindowText;
                      box-sizing: border-box;
                      color: WindowText;
                    }
                    & i {
                      font-weight: 600;
                    }
              >
                <span>
                  PA
                </span>
              </div>
            </div>
          </div>
          <div
            className=
                ms-Persona-details
                {
                  display: flex;
                  flex-direction: column;
                  justify-content: space-around;
                  min-width: 0px;
                  padding-bottom: 0;
                  padding-left: 8px;
                  padding-right: 8px;
                  padding-top: 0;
                  text-align: left;
                  width: 100%;
                }
          >
            <div
              className=
                  ms-Persona-primaryText
                  {
                    color: #005a9e;
                    font-size: 14px;
                    font-weight: 400;
                    overflow: hidden;
                    text-overflow: ellipsis;
                    white-space: nowrap;
                  }
                  @media screen and (-ms-high-contrast: active){& {
                    color: HighlightText;
                  }
              dir="auto"
            >
              <div
                className=
                    ms-TooltipHost
                    {
                      display: inline;
                    }
                onBlurCapture={[Function]}
                onFocusCapture={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                Person A
              </div>
            </div>
            <div
              className=
                  ms-Persona-secondaryText
                  {
                    color: #605e5c;
                    display: none;
                    font-size: 12px;
                    font-weight: 400;
                    overflow: hidden;
                    text-overflow: ellipsis;
                    white-space: nowrap;
                  }
              dir="auto"
            />
            <div
              className=
                  ms-Persona-tertiaryText
                  {
                    color: #605e5c;
                    display: none;
                    font-size: 14px;
                    font-weight: 400;
                    overflow: hidden;
                    text-overflow: ellipsis;
                    white-space: nowrap;
                  }
              dir="auto"
            />
            <div
              className=
                  ms-Persona-optionalText
                  {
                    color: #605e5c;
                    display: none;
                    font-size: 14px;
                    font-weight: 400;
                    overflow: hidden;
                    text-overflow: ellipsis;
                    white-space: nowrap;
                  }
              dir="auto"
            />
          </div>
        </div>
      </div>
      <button
        className=
            ms-Button
            ms-Button--icon
            ms-PickerItem-removeButton
            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              background-color: transparent;
              border-radius: 15px;
              border: none;
              box-sizing: border-box;
              color: #ffffff;
              cursor: pointer;
              display: inline-block;
              flex-basis: 32;
              flex: 0 0 auto;
              font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
              font-size: 14px;
              font-weight: 400;
              height: 33px;
              outline: transparent;
              padding-bottom: 0;
              padding-left: 4px;
              padding-right: 4px;
              padding-top: 0;
              position: relative;
              text-align: center;
              text-decoration: none;
              user-select: none;
              vertical-align: top;
              width: 33px;
            }
            &::-moz-focus-inner {
              border: 0;
            }
            .ms-Fabric--isFocusVisible &:focus:after {
              border: 1px solid transparent;
              bottom: 2px;
              content: "";
              left: 2px;
              outline: 1px solid #605e5c;
              position: absolute;
              right: 2px;
              top: 2px;
              z-index: 1;
            }
            @media screen and (-ms-high-contrast: active){.ms-Fabric--isFocusVisible &:focus:after {
              border: none;
              bottom: -2px;
              left: -2px;
              outline-color: ButtonText;
              right: -2px;
              top: -2px;
            }
            &:active > * {
              left: 0px;
              position: relative;
              top: 0px;
            }
            &:hover {
              background-color: #c7e0f4;
              color: #106ebe;
            }
            @media screen and (-ms-high-contrast: active){&:hover {
              border-color: Highlight;
              color: Highlight;
            }
            &:active {
              background-color: #edebe9;
              color: #005a9e;
            }
        data-is-focusable={true}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <div
          className=
              ms-Button-flexContainer
              {
                align-items: center;
                display: flex;
                flex-wrap: nowrap;
                height: 100%;
                justify-content: center;
              }
        >
          <i
            aria-hidden={true}
            className=
                ms-Button-icon
                {
                  -moz-osx-font-smoothing: grayscale;
                  -webkit-font-smoothing: antialiased;
                  color: #005a9e;
                  display: inline-block;
                  flex-shrink: 0;
                  font-family: "FabricMDL2Icons";
                  font-size: 16px;
                  font-style: normal;
                  font-weight: normal;
                  height: 16px;
                  line-height: 16px;
                  margin-bottom: 0;
                  margin-left: 4px;
                  margin-right: 4px;
                  margin-top: 0;
                  speak: none;
                  text-align: center;
                  vertical-align: middle;
                }
                @media screen and (-ms-high-contrast: active){& {
                  color: HighlightText;
                }
            data-icon-name="Cancel"
            role="presentation"
            style={
              Object {
                "fontSize": "14px",
              }
            }
          >
            Óúë
          </i>
        </div>
      </button>
    </div>
  </div>,
  <div
    aria-labelledby="selectedItemPersona-id__8"
    className=
        ms-PickerPersona-container
        {
          align-items: center;
          background: #eff6fc;
          border-radius: 15px;
          cursor: default;
          display: inline-flex;
          margin-bottom: 4px;
          margin-left: 4px;
          margin-right: 4px;
          margin-top: 4px;
          position: relative;
          user-select: none;
          vertical-align: middle;
        }
        &:hover {
          background: #deecf9;
        }
        @media screen and (-ms-high-contrast: active){&:hover {
          color: HighlightText;
        }
        @media screen and (-ms-high-contrast: active){& {
          -ms-high-contrast-adjust: none;
          background: Highlight;
          border-color: Highlight;
        }
    data-is-focusable={true}
    data-is-sub-focuszone={true}
    data-selection-index={1}
    role="listitem"
  >
    <div
      hidden={true}
    >
      <button
        className=
            ms-Button
            ms-Button--icon
            ms-PickerItem-removeButton
            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              background-color: transparent;
              border-radius: 15px 0px 0px 15px;
              border: none;
              box-sizing: border-box;
              color: #ffffff;
              cursor: pointer;
              display: flex;
              font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
              font-size: 14px;
              font-weight: 400;
              height: 33px;
              margin-right: -17px;
              outline: transparent;
              padding-bottom: 0;
              padding-left: 4px;
              padding-right: 16px;
              padding-top: 0;
              position: inherit;
              text-align: center;
              text-decoration: none;
              user-select: none;
              vertical-align: top;
              width: 44px;
            }
            &::-moz-focus-inner {
              border: 0;
            }
            .ms-Fabric--isFocusVisible &:focus:after {
              border: 1px solid transparent;
              bottom: 2px;
              content: "";
              left: 2px;
              outline: 1px solid #605e5c;
              position: absolute;
              right: 2px;
              top: 2px;
              z-index: 1;
            }
            @media screen and (-ms-high-contrast: active){.ms-Fabric--isFocusVisible &:focus:after {
              border: none;
              bottom: -2px;
              left: -2px;
              outline-color: ButtonText;
              right: -2px;
              top: -2px;
            }
            &:active > * {
              left: 0px;
              position: relative;
              top: 0px;
            }
            &:hover {
              background-color: #c7e0f4;
              color: #106ebe;
            }
            @media screen and (-ms-high-contrast: active){&:hover {
              border-color: Highlight;
              color: Highlight;
            }
            &:active {
              background-color: #edebe9;
              color: #005a9e;
            }
        data-is-focusable={true}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <div
          className=
              ms-Button-flexContainer
              {
                align-items: center;
                display: flex;
                flex-wrap: nowrap;
                height: 100%;
                justify-content: center;
              }
        >
          <i
            aria-hidden={true}
            className=
                ms-Button-icon
                {
                  -moz-osx-font-smoothing: grayscale;
                  -webkit-font-smoothing: antialiased;
                  color: #005a9e;
                  display: inline-block;
                  flex-shrink: 0;
                  font-family: "FabricMDL2Icons";
                  font-size: 16px;
                  font-style: normal;
                  font-weight: normal;
                  height: 16px;
                  line-height: 16px;
                  margin-bottom: 0;
                  margin-left: 4px;
                  margin-right: 4px;
                  margin-top: 0;
                  speak: none;
                  text-align: center;
                  vertical-align: middle;
                }
                @media screen and (-ms-high-contrast: active){& {
                  color: HighlightText;
                }
            data-icon-name="Add"
            role="presentation"
            style={
              Object {
                "fontSize": "14px",
              }
            }
          >
            Óúê
          </i>
        </div>
      </button>
    </div>
    <div
      className=

          {
            display: inherit;
            position: relative;
          }
    >
      <div
        className=
            ms-PickerItem-content
            {
              flex: 0 1 auto;
              max-width: 100%;
              min-width: 0px;
            }
        id="selectedItemPersona-id__8"
      >
        <div
          className=
              ms-Persona
              ms-Persona--size32
              {
                -moz-osx-font-smoothing: grayscale;
                -webkit-font-smoothing: antialiased;
                align-items: center;
                box-shadow: none;
                box-sizing: border-box;
                color: #323130;
                display: flex;
                font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                font-size: 14px;
                font-weight: 400;
                height: 32px;
                margin-bottom: 0px;
                margin-left: 0px;
                margin-right: 0px;
                margin-top: 0px;
                min-width: 32px;
                padding-bottom: 0px;
                padding-left: 0px;
                padding-right: 0px;
                padding-top: 0px;
                position: relative;
              }
              & .contextualHost {
                display: none;
              }
              &:hover $primaryText {
                color: #201f1e;
              }
        >
          <div
            className=
                ms-Persona-coin
                ms-Persona--size32
                {
                  -moz-osx-font-smoothing: grayscale;
                  -webkit-font-smoothing: antialiased;
                  font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                  font-size: 14px;
                  font-weight: 400;
                }
          >
            <div
              className=
                  ms-Persona-imageArea
                  {
                    flex: 0 0 auto;
                    height: 32px;
                    position: relative;
                    text-align: center;
                    width: 32px;
                  }
            >
              <div
                aria-hidden="true"
                className=
                    ms-Persona-initials
                    {
                      background-color: #CA5010;
                      border-radius: 50%;
                      color: #ffffff;
                      font-size: 14px;
                      font-weight: 600;
                      height: 32px;
                      line-height: 32px;
                    }
                    @media screen and (-ms-high-contrast: active){& {
                      -ms-high-contrast-adjust: none;
                      background-color: Window !important;
                      border: 1px solid WindowText;
                      box-sizing: border-box;
                      color: WindowText;
                    }
                    & i {
                      font-weight: 600;
                    }
              >
                <span>
                  PB
                </span>
              </div>
            </div>
          </div>
          <div
            className=
                ms-Persona-details
                {
                  display: flex;
                  flex-direction: column;
                  justify-content: space-around;
                  min-width: 0px;
                  padding-bottom: 0;
                  padding-left: 8px;
                  padding-right: 8px;
                  padding-top: 0;
                  text-align: left;
                  width: 100%;
                }
          >
            <div
              className=
                  ms-Persona-primaryText
                  {
                    color: #005a9e;
                    font-size: 14px;
                    font-weight: 400;
                    overflow: hidden;
                    text-overflow: ellipsis;
                    white-space: nowrap;
                  }
                  @media screen and (-ms-high-contrast: active){& {
                    color: HighlightText;
                  }
              dir="auto"
            >
              <div
                className=
                    ms-TooltipHost
                    {
                      display: inline;
                    }
                onBlurCapture={[Function]}
                onFocusCapture={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                Person B
              </div>
            </div>
            <div
              className=
                  ms-Persona-secondaryText
                  {
                    color: #605e5c;
                    display: none;
                    font-size: 12px;
                    font-weight: 400;
                    overflow: hidden;
                    text-overflow: ellipsis;
                    white-space: nowrap;
                  }
              dir="auto"
            />
            <div
              className=
                  ms-Persona-tertiaryText
                  {
                    color: #605e5c;
                    display: none;
                    font-size: 14px;
                    font-weight: 400;
                    overflow: hidden;
                    text-overflow: ellipsis;
                    white-space: nowrap;
                  }
              dir="auto"
            />
            <div
              className=
                  ms-Persona-optionalText
                  {
                    color: #605e5c;
                    display: none;
                    font-size: 14px;
                    font-weight: 400;
                    overflow: hidden;
                    text-overflow: ellipsis;
                    white-space: nowrap;
                  }
              dir="auto"
            />
          </div>
        </div>
      </div>
      <button
        className=
            ms-Button
            ms-Button--icon
            ms-PickerItem-removeButton
            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              background-color: transparent;
              border-radius: 15px;
              border: none;
              box-sizing: border-box;
              color: #ffffff;
              cursor: pointer;
              display: inline-block;
              flex-basis: 32;
              flex: 0 0 auto;
              font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
              font-size: 14px;
              font-weight: 400;
              height: 33px;
              outline: transparent;
              padding-bottom: 0;
              padding-left: 4px;
              padding-right: 4px;
              padding-top: 0;
              position: relative;
              text-align: center;
              text-decoration: none;
              user-select: none;
              vertical-align: top;
              width: 33px;
            }
            &::-moz-focus-inner {
              border: 0;
            }
            .ms-Fabric--isFocusVisible &:focus:after {
              border: 1px solid transparent;
              bottom: 2px;
              content: "";
              left: 2px;
              outline: 1px solid #605e5c;
              position: absolute;
              right: 2px;
              top: 2px;
              z-index: 1;
            }
            @media screen and (-ms-high-contrast: active){.ms-Fabric--isFocusVisible &:focus:after {
              border: none;
              bottom: -2px;
              left: -2px;
              outline-color: ButtonText;
              right: -2px;
              top: -2px;
            }
            &:active > * {
              left: 0px;
              position: relative;
              top: 0px;
            }
            &:hover {
              background-color: #c7e0f4;
              color: #106ebe;
            }
            @media screen and (-ms-high-contrast: active){&:hover {
              border-color: Highlight;
              color: Highlight;
            }
            &:active {
              background-color: #edebe9;
              color: #005a9e;
            }
        data-is-focusable={true}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <div
          className=
              ms-Button-flexContainer
              {
                align-items: center;
                display: flex;
                flex-wrap: nowrap;
                height: 100%;
                justify-content: center;
              }
        >
          <i
            aria-hidden={true}
            className=
                ms-Button-icon
                {
                  -moz-osx-font-smoothing: grayscale;
                  -webkit-font-smoothing: antialiased;
                  color: #005a9e;
                  display: inline-block;
                  flex-shrink: 0;
                  font-family: "FabricMDL2Icons";
                  font-size: 16px;
                  font-style: normal;
                  font-weight: normal;
                  height: 16px;
                  line-height: 16px;
                  margin-bottom: 0;
                  margin-left: 4px;
                  margin-right: 4px;
                  margin-top: 0;
                  speak: none;
                  text-align: center;
                  vertical-align: middle;
                }
                @media screen and (-ms-high-contrast: active){& {
                  color: HighlightText;
                }
            data-icon-name="Cancel"
            role="presentation"
            style={
              Object {
                "fontSize": "14px",
              }
            }
          >
            Óúë
          </i>
        </div>
      </button>
    </div>
  </div>,
]
`;
